*: remove `__func__` from debug messages

eqvinox opened the pull request:
Since we already know the function name in the logging code, we can shave off some thousand occurences of `__func__` in the code.

Includes a warning printed by `xrelfo.py` if `__func__` is in a debug message (does not get triggered on Linux, CI will tell me about BSDs…)

eqvinox committed:
doc/developer: exclude function name from debugs

Policy change/documentation on excluding function name on debug logs.

Signed-off-by: David Lamparter <equinox@opensourcerouting.org>

eqvinox committed:
lib: include function name on debug logs

Signed-off-by: David Lamparter <equinox@opensourcerouting.org>

eqvinox committed:
*: remove `__func__` from zlog_debug calls

Also includes zlog_debug-"adjacent" debug calls.

Result formatted with clang-format.

Signed-off-by: David Lamparter <equinox@opensourcerouting.org>

eqvinox committed:
tools: add `__func__` removal tool

This does the search & replace for the next commit.

Signed-off-by: David Lamparter <equinox@opensourcerouting.org>

eqvinox committed:
python: enable `__func__` warning in xrelfo

Prints a warning for debug messages starting with `__func__` as their
first arg.

Signed-off-by: David Lamparter <equinox@opensourcerouting.org>

eqvinox committed:
*: remove `__func__` from debug macros

Signed-off-by: David Lamparter <equinox@opensourcerouting.org>

eqvinox committed:
*: manual remove `__func__` in debug messages

Signed-off-by: David Lamparter <equinox@opensourcerouting.org>

donaldsharp commented:
Talking w/ @eqvinox via slack and I asked about output for zlog_hexdump or the re dumper from zebra_rib.c and got this output:
```2022/02/25 14:59:29.124875 debugging: ZEBRA: [N7TEE-83DGB] rib_sweep_table: starting
2022/02/25 14:59:29.124876 debugging: ZEBRA: [Q4T2G-E2SQF] _route_entry_dump: rib_sweep_table: dumping RE entry 0x5565a82fab00 for fe80::/64 vrf default(0)
2022/02/25 14:59:29.124876 debugging: ZEBRA: [M5M58-9PD2R] _route_entry_dump: fe80::/64: uptime == 24584, type == 2, instance == 0, table == 254
2022/02/25 14:59:29.124877 debugging: ZEBRA: [RVZMM-N7DME] _route_entry_dump: fe80::/64: metric == 0, mtu == 0, distance == 0, flags == Selected status == None 
2022/02/25 14:59:29.124878 debugging: ZEBRA: [Q1NW5-NWY7P] _route_entry_dump: fe80::/64: nexthop_num == 1, nexthop_active_num == 1
2022/02/25 14:59:29.124878 debugging: ZEBRA: [TFHQ8-TC30H] _route_entry_dump_nh: fe80::/64: NH wlp1s0[4] vrf default(0)  with flags ACTIVE 
2022/02/25 14:59:29.124879 debugging: ZEBRA: [SCETK-GQ9E4] _route_entry_dump: fe80::/64: dump complete
2022/02/25 14:59:29.124880 debugging: ZEBRA: [VMA1Q-6C53X] rib_sweep_table: ends``` 

I believe David is going to look into a way to allow us to specify in some cases the calling functions name to better handle this situation

eqvinox commented:
ack, going to add some way to "tack on" / pass through parent function location for helpers like that.

mjstapp reviewed (commented):
hmm - getting the function name for free seems nice, but the impact of changing every existing debug seems a little out of scale to me. how about adding a variant like `zlog_debugf` or `zlog_debugx` that can be used going forward instead? I'm concerned about problems with, for instance, things like backports.

donaldsharp commented:
I would prefer we make the hard changes instead of having another round of partially done work :(

mjstapp commented:
So I don't regard it as "partially done" - I think of it as an improvement to logging, that becomes available for use. There's nothing wrong with the improvement; it's an improvement. Churning thousands of lines of code every time we add something useful isn't worthwhile.

> I would prefer we make the hard changes instead of having another round of partially done work :(

donaldsharp commented:
I think it is because we other wise end up with 5 different ways to do the same thing, which is very confusing.  And we have already done the same thing for log messaging and `%pXXX` codes and a myriad of other things.  Why are you drawing the line here?

eqvinox committed:
lib: add zlog_blk{_begin,_end,_debug}

These allow grouping debug messages together.  Future code may aggregate
them into one multi-line log message.  For now, this only affects the
function name printed.

Signed-off-by: David Lamparter <equinox@opensourcerouting.org>

eqvinox committed:
zebra: use zlog_blk in route_entry_dump()

Signed-off-by: David Lamparter <equinox@opensourcerouting.org>

eqvinox committed:
lib: use zlog_blk in zlog_hexdump()

Signed-off-by: David Lamparter <equinox@opensourcerouting.org>

riw777 commented:
@frrbot autoclose in 1 month

eqvinox commented:
This is coming back, but it's gonna be a fresh PR (it's gone through some evolution anyway.)

Closing.
